packages/mobile/src/identity/contactMapping.ts

```diff
@@ -96,11 +104,15 @@ function* updateRecipientsCache(
   yield put(setRecipientCache({ ...e164NumberToRecipients, ...otherRecipients }))
 }
 
-export function* fetchPhoneAddresses({ e164Number }: FetchPhoneAddressesAction) {
+export function* fetchAddressesAndValidateSaga({ e164Number }: FetchAddressesAndValidateAction) {
   try {
-    Logger.debug(TAG + '@fetchPhoneAddresses', `Fetching addresses for number`)
-    // Clear existing entries for those numbers so our mapping consumers
-    // know new status is pending.
+    Logger.debug(TAG + '@fetchAddressesAndValidate', `Fetching addresses for number`)
+    const oldE164NumberToAddress = yield select(e164NumberToAddressSelector)
+    const oldAddresses = oldE164NumberToAddress[e164Number]
```

Comment:
Tip: these 3 lines could be:
`const oldAddresses = oldE164NumberToAddress[e164Number] || []`

Reply:
That wasn't what I was originally thinking because `oldE164NumberToAddress[e164Number]` could evaluate to `null` or `undefined` as well. But, I could evaluate those to equal `[]` instead and handle it differently downstream